src/object/endpoint.c

```diff
@@ -226,10 +226,10 @@ void receiveIPC(tcb_t *thread, cap_t cap, bool_t isBlocking)
             do_call = thread_state_ptr_get_blockingIPCIsCall(&sender->tcbState);
 
 #ifdef CONFIG_KERNEL_MCS
-            if (do_call ||
-                seL4_Fault_get_seL4_FaultType(sender->tcbFault) != seL4_Fault_NullFault) {
+            word_t faultType = seL4_Fault_get_seL4_FaultType(sender->tcbFault);
+            if (do_call || faultType != seL4_Fault_NullFault) {
                 if ((canGrant || canGrantReply) && replyPtr != NULL) {
-                    reply_push(sender, thread, replyPtr, sender->tcbSchedContext != NULL);
+                    reply_push(sender, thread, replyPtr, faultType != seL4_Fault_Timeout);
```

Comment:
I'm confused -- why remove `sender->tcbSchedContext != null`?

Reply:
`reply_push` performs that check internally. The fourth argument is only used in a condition where it combined with a logical and of the sender's SC not being `NULL`.